Sources/Vehicle/Extensions/HAEntityExtension.swift

```diff
@@ -0,0 +1,215 @@
+import Foundation
+import HAKit
+import PromiseKit
+import Shared
+import SwiftUI
+
+extension HAEntity {
+    public static func getIconForDomain(domain: String, size: CGSize) -> UIImage? {
+        do {
+            let tmpEntity = try HAEntity(entityId: "\(domain).ha_ios_placeholder",
+                                         domain: domain, state: "",
+                                         lastChanged: Date(),
+                                         lastUpdated: Date(),
+                                         attributes: [:],
+                                         context: HAResponseEvent.Context(id: "", userId: nil, parentId: nil))
+            return tmpEntity.getIcon(size: size)
+        }
+        catch {
+            return nil
+        }
+    }
+    
+    func onPress(for api: HomeAssistantAPI) -> Promise<Void>{
+        let domain = domain
+        var service: String
+        switch (domain) {
+            case "lock":
+                service = state == "unlocked" ? "lock" : "unlock"
+            case "cover":
+                service = state == "open" ? "close_cover" : "open_cover"
+            case "button","input_button":
+                service = "press";
+            case "scene":
+                service = "turn_on";
+            default:
+                service = state == "on" ? "turn_off" : "turn_on";
+        }
+        return api.CallService(domain: domain, service: service, serviceData: ["entity_id": entityId])
```

Comment:
you're mixing websocket code with webhook code; you should continue down the websocket path. create a HATypedRequest extension that calls the service, and use that strongly-typed value to do the service calls above, rather than stringifying everything.

Reply:
Can you provide an example with your suggestion? How can I use HATypedRequest to make a state change request for turn_off?

Reply:
I have added the typed requests